logevents/src/main/java/org/logevents/observers/ElasticsearchLogEventObserver.java

```diff
@@ -34,34 +40,93 @@
  * observer.elastic.suppressMarkers=PERSONAL_DATA
  * observer.elastic.formatter.excludedMdcKeys=secret
  * </pre>
+ * Optional configuration shown below, but with their defaults
+ * <pre>
+ * observer.elastic.elasticsearchUrlPath=_bulk
+ * </pre>
+ * And you can provide Authorization header with
+ * <pre>
+ * observer.elastic.elasticsearchAuthorizationHeader
+ * </pre>
  */
 public class ElasticsearchLogEventObserver extends AbstractBatchingLogEventObserver {
 
+    public static final String DEFAULT_ELASTICSEARCH_BULK_API_PATH = "_bulk";
     private final URL elasticsearchUrl;
+    protected final String elasticsearchUrlPath;
+    private final String elasticsearchAuthorizationHeaderValue;
     private final String index;
     private final JsonLogEventFormatter formatter;
 
+    private Proxy proxy = Proxy.NO_PROXY;
+
     public ElasticsearchLogEventObserver(Map<String, String> properties, String prefix) {
         this(new Configuration(properties, prefix));
     }
 
     public ElasticsearchLogEventObserver(Configuration configuration) {
         this.elasticsearchUrl = configuration.getUrl("elasticsearchUrl");
+        this.elasticsearchUrlPath = configuration.optionalString("elasticsearchUrlPath").orElse(getDefaultPath());
+        this.elasticsearchAuthorizationHeaderValue = configuration.optionalString("elasticsearchAuthorizationHeader").orElse(NO_AUTHORIZATION_HEADER);
         this.index = configuration.getString("index");
         this.formatter = configuration.createInstanceWithDefault("formatter", JsonLogEventFormatter.class);
         this.configureBatching(configuration);
         this.configureFilter(configuration, Level.TRACE);
+        this.configureMarkers(configuration);
+        this.configureProxy(configuration);
         configuration.checkForUnknownFields();
     }
 
-    public ElasticsearchLogEventObserver(URL elasticsearchUrl, String index) {
-        this.elasticsearchUrl = elasticsearchUrl;
-        this.index = index;
-        this.formatter = new JsonLogEventFormatter();
+    protected String getDefaultPath() {
+        return DEFAULT_ELASTICSEARCH_BULK_API_PATH;
+    }
+
+    protected static class ElasticsearchPropertiesConfigurationBuilder {
```

Comment:
It is used by the more convinent constructors here: https://github.com/norrs/logevents/blob/humio/logevents/src/main/java/org/logevents/observers/ElasticsearchLogEventObserver.java#L108 

However, I'll move the constructor to the top of the class. Its protected so it could be used by the extending class and easily from the unit tests.

Reply:
But I see in in the slack observer test that you prefer to build a `Map<String, String>` and pass it in. I can do this instead of using the builder pattern.

Reply:
I can move the builder to the test class instead?

Reply:
What is the effect you're trying to achieve by having a builder?

Reply:
It has a convinent and safe way for providing the configuration values without typos. Useful if constructing it from code and the constructor has more than 2> arguments as java is missing named parameters. 

So something like:

```
public ElasitcsearchLogEventObserver {
  protected ElasitcsearchLogEventObserver(Builder b) {
    // create Map<String, String> without typos on property names
  } 
  
  public static class Builder {
      public Builder() { }
      
      public withElasticsearchUrl(URL url) { (do we really want to use URL class? its rather broken class, always used URI hm)
        this.eUrl = url
        return this
      }
      ...
      public build() {
        new ElasitcsearchLogEventObserver(this)
      }
      
  }

```